android/src/main/java/com/oney/WebRTCModule/UVCCamera2Enumerator.java

```diff
@@ -0,0 +1,101 @@
+package com.oney.WebRTCModule;
+
+import android.content.Context;
+import android.hardware.usb.UsbDevice;
+
+import androidx.annotation.Nullable;
+
+import com.jiangdg.ausbc.camera.CameraUvcStrategy;
+import com.jiangdg.ausbc.camera.bean.PreviewSize;
+
+import org.webrtc.Camera2Enumerator;
+import org.webrtc.CameraEnumerationAndroid;
+import org.webrtc.CameraVideoCapturer;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+/**
+ * Camera2Enumerator with UVC Camera support.
+ * <p/>
+ * This implementation only adds support for UVC camera devices.
+ * Ie. Device's (back and/or front) cameras will use the basic implementation of {@link Camera2Enumerator}.
+ */
+public class UVCCamera2Enumerator extends Camera2Enumerator  {
+
+    /** UVC camera names will be prefix with this value. Currently, there is no other way to
+     * easily distinguish between device's own and external uvc cameras in JS-side. */
+    public static final String UVC_PREFIX = "uvc-camera:";
+
+    private final UVCMonitor uvcMonitor;
+    private final Context context;
+
+    public UVCCamera2Enumerator(Context context) {
+        super(context);
+        this.uvcMonitor = new UVCMonitor(context);
+        this.context = context;
+    }
+
+    @Override
+    public String[] getDeviceNames() {
+        ArrayList<String> devicesNames = new ArrayList<>(Arrays.asList(super.getDeviceNames()));
+        UsbDevice uvcDevice = this.uvcMonitor.getUvcCameraDevice();

```

Comment:
We could get the uvc device from the strategy itself, but it has certain weaknesses. Ie: The camera needs to be registered and will need a permission from the user, which IMO should only be needed when user wants to stream the camera).

Thus, i created a wrapper (UVCMonitor), which can provide UVC device without prompting permission from the user.

Reply:
You mean enumerating the devices would require permissions?

Reply:
Yes, we would need to request a permission to use the attached UVC device in order to even enumerate it.

This approach (where permission is asked when actual streaming is started) is much more user friendly imo.

Reply:
I see. Users would need to enumerate the device first to know it's there, wouldn't they?

Reply:
I think i explained this a bit confusingly, sorry.

I'll try again,
CameraUvcStrategy (which we use in the capturer) has a getCurrentDevice(): UsbDevice method, which could be used for enumerating purposes. But, in order to use that method we would need a runtime permission from the user.
By using the UVCMonitor, no runtime- or app permissions are required for the enumeration part.

Reply:
Understood now, that sounds great!